rotkehlchen/chain/evm/decoding/pendle/decoder.py

```diff
@@ -280,6 +294,149 @@ def _decode_pendle_swap(self, context: DecoderContext) -> DecodingOutput:
         )
         return DecodingOutput(action_items=action_items)
 
+    def _decode_mint_sy_pt_yt_from_token(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes Pendle deposits where an underlying token is converted into SY, PT, or YT."""
+        amount_out = asset_normalized_value(
+            amount=int.from_bytes(context.tx_log.data[32:64]),
+            asset=(token_out := self.base.get_token_or_native(bytes_to_address(context.tx_log.topics[2]))),  # noqa: E501
+        )
+        amount_in = asset_normalized_value(
+            amount=int.from_bytes(context.tx_log.data[64:96]),
+            asset=self.base.get_token_or_native(bytes_to_address(context.tx_log.topics[3])),
+        )
+
+        out_event, in_events = None, []
+        for event in context.decoded_events:
+            if (
+                    event.amount == amount_out and
+                    event.asset == token_out and
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+            ):
+                out_event = event
+                event.counterparty = CPT_PENDLE
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_FOR_WRAPPED
+                event.notes = f'Deposit {event.amount} {token_out.symbol} to Pendle'
+
+            elif (
+                    # Pendle emits one RECEIVE per output token (SY, PT, YT),
+                    # each with the same amount.
+                    event.amount == amount_in and
+                    event.address == ZERO_ADDRESS and
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE
+            ):
+                in_events.append(event)
+                event.counterparty = CPT_PENDLE
+                event.event_type = HistoryEventType.RECEIVE
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.notes = f'Receive {event.amount} {event.asset.resolve_to_asset_with_symbol().symbol} from depositing into Pendle'  # noqa: E501
+
+        if out_event is None or len(in_events) == 0:
+            log.error(f'Could not retrieve either out event or in events in pendle mint sy/pt/yt transaction {context.transaction}')  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        maybe_reshuffle_events(
+            ordered_events=[out_event] + in_events,
+            events_list=context.decoded_events,
+        )
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_redeem_sy_pt_yt_to_token(self, context: DecoderContext) -> DecodingOutput:
```

Comment:
this function and `_decode_mint_sy_pt_yt_from_token` seem to have very very similar logic? Please abstract them. Only few things are different between them